### ISSUE TITLE
Return if a solving is successful or not, instead of throwing UnsolvedSymbolException

### ISSUE THREAD

**Issue by Symbolk** [javaparser/javaparser#1992]
In gradle/maven projects, depending jars are usually downloaded in building, so we do not have the jarFolder to set up the JarTypeSolver:

```java
    //    TypeSolver jarTypeSolver = JarTypeSolver.getJarTypeSolver(jarFolder);
    TypeSolver reflectionTypeSolver = new ReflectionTypeSolver();
    TypeSolver javaParserTypeSolver = new JavaParserTypeSolver(new File(projectSrcFolder));
    reflectionTypeSolver.setParent(reflectionTypeSolver);
    CombinedTypeSolver combinedTypeSolver = new CombinedTypeSolver();
    combinedTypeSolver.add(reflectionTypeSolver);
    combinedTypeSolver.add(javaParserTypeSolver);
    JavaSymbolSolver symbolSolver = new JavaSymbolSolver(combinedTypeSolver);
    JavaParser.getStaticConfiguration().setSymbolResolver(symbolSolver);
```
Right now I am using JavaSymbolSolver to resolve all MethodCallExpr and FieldAccessExpr in one class. I want to resolve ONLY the "internal" method call or field access. By saying "internal", I mean those methods or fields that are defined in the java files of the current project, instead of in jars. So I set up the JavaParser like above, the result is UnsolvedSymbolException{...} are thrown when solving methods or fields in jars.

I have tried this way, but still the case:
```java
    final JavaParserFacade javaParserFacade = JavaParserFacade.get(combinedTypeSolver);
            SymbolReference<ResolvedMethodDeclaration> ref = javaParserFacade.solve((methodCallExpr));
if(ref.isSolved()){
....
}
```

In summary, I wonder how to:
1.  judge whether a method call is internal with JavaParser or in other way?
2.  jump the failed solving and continue solving the next.

Thanks for any help!

**Comment by ftomassetti** [comment]
Well, the fact is that often resolving somethign is a prerequisite to resolving something else. For example, if you have a method chain:

```
a().b().c()
```

You need to know what `a()` returns to figure out what `b` corresponds to. So "eating" unsolved symbols and ignoring them is dangerous.

What we can know if something can be resolved with the internal resources we know about (basically the code we have + reflection) or not. In the latter case we can assume they are references to JARs we do now have.

Anyway I would basically create a loop in which I catch the UnsolvedSymbolException and log it, moving to the next one. I am not sure if this is enough or you wanted more. In that case please give me more details.

**Comment by Symbolk** [comment]
> Well, the fact is that often resolving somethign is a prerequisite to resolving something else. For example, if you have a method chain:
> 
> ```
> a().b().c()
> ```
> You need to know what `a()` returns to figure out what `b` corresponds to. So "eating" unsolved symbols and ignoring them is dangerous.
> 
> What we can know if something can be resolved with the internal resources we know about (basically the code we have + reflection) or not. In the latter case we can assume they are references to JARs we do now have.
> 
> Anyway I would basically create a loop in which I catch the UnsolvedSymbolException and log it, moving to the next one. I am not sure if this is enough or you wanted more. In that case please give me more details.

Exactly, what I did is to wrap the solving process in try block, and in case of catching UnsolvedSymbolException, just continue to try solving the next one.

**Comment by KRK224** [comment]
Hello,
I've got a same problem as him, is there any solutions?
